frontend/src/quizView/quizView.js

```diff
@@ -0,0 +1,287 @@
+                question: nextQuestionNumber,
+            });
+        } else {
+            console.log();
+        }
+    }
+
+    prevQuestion() {
+        const prevQuestionNumber = this.state.question - 1;
+        if (prevQuestionNumber > 0) {
+            this.setState({
+                question: prevQuestionNumber,
+            });
+        } else {
+            console.log();
+        }
+    }
+
+    getUnanswered() {
+        let numUnanswered = 0;
+        const answers = this.state.userAnswers;
+        for (let i = 1; i <= this.state.data.length; i++) {
+            if (answers[i] === undefined) {
+                numUnanswered++;
+            }
+        }
+        return numUnanswered;
+    }
+
+    gradeQuiz() {
+        let cSubmit;
+        if (this.getUnanswered() === 0) {
+            // eslint-disable-next-line no-alert
+            cSubmit = window.confirm(
+                'Are you sure that you want to submit your quiz? Your answers are final.',
+            );
+        } else {
+            // eslint-disable-next-line no-alert
+            cSubmit = window.confirm(
+                'Are you sure that you want to submit your quiz? '
+                + `You have ${this.getUnanswered()} unanswered questions(s).`,
+            );
+        }
+        if (cSubmit === true) {
+            this.setState({ graded: true });
+            let score = 0;
+            const answers = this.state.userAnswers;
+            for (let i = 0; i < this.state.data.length; i++) {
+                if (answers[i + 1] === this.state.data[i].answer) {
+                    score += 1;
+                }
+            }
+            this.setState({ score: score });
+        } else {
+            console.log();
+        }
+    }
+
+    onProgressBarClick = (event) => {
+        this.setState({
+            ...this.state,
+            question: parseInt(event.target.value),
+        });
+    }
+
+    onAnswerChoiceClick = (event) => {
+        const currentQ = this.state.question;
+        if ((typeof event.target.value) === 'undefined') {
+            console.log();
+        } else if (!this.state.graded) {
+            const answers = this.state.userAnswers;
+            answers[currentQ] = event.target.value;
+            this.setState({ userAnswers: answers });
+        }
+    }
+
+    render() {
+        if (!this.state.data) {
+            return (<LoadingPage />);
+        }
+
+        // Previously a ButtonChoices function
+        const choices = this.state.data[this.state.question - 1].options;
+
+        const radios = choices.map((choice, i) => {
+            return (
+                <ToggleButton
+                    key={i + 1}
+                    variant="outline-light"
+                    value={choice}
+                    className="c-button"
+                    onClick={this.onAnswerChoiceClick}
+                >
+                    {choice}
+                </ToggleButton>
+            );
+        });
+
+        return (
```

Comment:
See my comment from review of #9 about splitting this up into some helper functions.